### PR TITLE
API call to get files from a container/volume

### PR DESCRIPTION
**Pull request by ubuntolog**
Uses Docker remote API directly

### CODE REVIEW

**Comment by ubuntolog** [comment]
Travis crashed while building the frontend, although I have not touched the frontend

**Comment by ubuntolog** [comment]
I need to move a couple of constants to the config file

File: backend-docker/server/apiserver.go [hunk 1]
```diff
@@ -38,10 +39,14 @@ const (
 	buildImagesAPIPath   = "/buildImages"
 	buildVolumesAPIPath  = "/buildVolumes"
 	inspectVolumeAPIPath = "/inspectVolume"
+	retrieveFileAPIPath = "/retrieveFile"
 
 	tmpDirDefault = "gefdocker"
 	tmpDirPerm    = 0700
 	buildsTmpDir  = "builds"
+
+	sock = "/var/run/docker.sock"
```
Comment (emanueldima) [inline backend-docker/server/apiserver.go]: Yep, this should be in the config file

File: backend-docker/server/apiserver.go [hunk 2]
```diff
@@ -123,8 +129,30 @@ func NewServer(cfg Config, docker dckr.Client) *Server {
 	return server
 }
 
+// remoteApiServer starts a server needed to access Docker remote API
```
Comment (emanueldima) [inline backend-docker/server/apiserver.go]: No, please use the docker client library that we already have, let's not reimplement the API.

````
func (c *Client) DownloadFromContainer(id string, opts DownloadFromContainerOptions) error
````

see https://github.com/fsouza/go-dockerclient/blob/master/container.go#L1200

**Comment by emanueldima** [comment]
build fails!

**Comment by ubuntolog** [comment]
It is green now

File: backend-docker/server/apiserver.go (hunk 1 above)
Comment (emanueldima) [inline backend-docker/server/apiserver.go]: Ok, this is not unused, can be removed.

**Comment by emanueldima** [comment]
Please rebase on master and then we can merge

File: backend-docker/server/apiserver.go (hunk 1 above)
Comment (ubuntolog) [inline backend-docker/server/apiserver.go]: Right

**Comment by ubuntolog** [comment]
Done

**Pull request merged**